core/src/include/ModelMetadata.hpp

```diff
@@ -18,6 +18,9 @@
 #ifdef USE_MPI
 #include <mpi.h>
 #endif
+#ifdef USE_OASIS
```

Comment:
The inclusion of OASIS in a `.hpp` included by main programs, require to add for many `XXX` targets in `core/test/CMakeLists.txt` and in `physics/test/CMakeLists.txt` a bunch of sections like  
```
if (ENABLE_OASIS)
   target_include_directories(XXX PRIVATE ${OASIS_INCLUDES})
endif ()
```

Reply:
Is this working better now that I put ``PUBLIC`` in the root ``CMakeLists.txt``? I don't have any problems here.

Reply:
It's working allright with only two other changes:

1. Rename `Findoasis.cmake` to `FindOASIS.cmake` (my version seems to be case sensitive)
2. Replace `${OASIS_DIR}/inc` by `${OASIS_DIR}/include` in it

Reply:
Updated with the latest commit

1. I don't quite understand why the case matters, but my setup appears to be case-insensitive
2. It should have been ``include`` all along. It's a bit of a mystery why it worked for me the way it was!